### PR TITLE
Update xboxkrnl_modules to new convention

### PR DESCRIPTION
**Pull request by maxton**
See #631

### CODE REVIEW

File: src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc [hunk 1]
```diff
@@ -101,131 +101,102 @@ X_STATUS xeExGetXConfigSetting(uint16_t category, uint16_t setting,
...
-  XELOGD("XexGetModuleSection(%.8X, %s, %.8X, %.8X)", hmodule, name, data_ptr,
-         size_ptr);
-
+dword_result_t XexGetModuleSection(lpvoid_t hmodule, lpstring_t name,
+                                   lpdword_t data_ptr, lpdword_t size_ptr) {
   X_STATUS result = X_STATUS_SUCCESS;
 
-  auto module = XModule::GetFromHModule(kernel_state, SHIM_MEM_ADDR(hmodule));
+  auto module =
+      XModule::GetFromHModule(kernel_state(), (void*)hmodule.host_address());
```
Comment (DrChat) [inline src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc]: Pass hmodule itself as the void pointer, no need to cast the host address.

File: src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc [hunk 2]
```diff
@@ -101,131 +101,102 @@ X_STATUS xeExGetXConfigSetting(uint16_t category, uint16_t setting,
...
+dword_result_t XexGetModuleHandle(lpstring_t module_name,
+                                  lpdword_t hmodule_ptr) {
   object_ref<XModule> module;
+
   if (!module_name) {
-    module = kernel_state->GetExecutableModule();
+    module = kernel_state()->GetExecutableModule();
   } else {
-    module = kernel_state->GetModule(module_name);
+    module = kernel_state()->GetModule((char*)module_name);
```
Comment (DrChat) [inline src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc]: No need to cast module_name to char*

File: src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc [hunk 3]
```diff
@@ -240,65 +211,53 @@ SHIM_CALL XexLoadImage_shim(PPCContext* ppc_context,
...
-SHIM_CALL XexUnloadImage_shim(PPCContext* ppc_context,
-                              KernelState* kernel_state) {
-  uint32_t hmodule = SHIM_GET_ARG_32(0);
-
-  XELOGD("XexUnloadImage(%.8X)", hmodule);
-
-  auto module = XModule::GetFromHModule(kernel_state, SHIM_MEM_ADDR(hmodule));
+dword_result_t XexUnloadImage(lpvoid_t hmodule) {
+  auto module =
+      XModule::GetFromHModule(kernel_state(), (void*)hmodule.host_address());
```
Comment (DrChat) [inline src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc]: Same as the last comment.

File: src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc [hunk 4]
```diff
@@ -240,65 +211,53 @@ SHIM_CALL XexLoadImage_shim(PPCContext* ppc_context,
...
+    return X_STATUS_INVALID_HANDLE;
   }
 
   // Can't unload kernel modules from user code.
   if (module->module_type() != XModule::ModuleType::kKernelModule) {
-    auto ldr_data =
-        kernel_state->memory()->TranslateVirtual<X_LDR_DATA_TABLE_ENTRY*>(
-            hmodule);
+    auto ldr_data = kernel_memory()->TranslateVirtual<X_LDR_DATA_TABLE_ENTRY*>(
+        hmodule.guest_address());
```
Comment (DrChat) [inline src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc]: auto ldr_data = hmodule.as<X_LDR_...>();

File: src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc [hunk 5]
```diff
@@ -240,65 +211,53 @@ SHIM_CALL XexLoadImage_shim(PPCContext* ppc_context,
...
   X_STATUS result = X_STATUS_INVALID_HANDLE;
 
   object_ref<XModule> module;
   if (!hmodule) {
-    module = kernel_state->GetExecutableModule();
+    module = kernel_state()->GetExecutableModule();
   } else {
-    module = XModule::GetFromHModule(kernel_state, SHIM_MEM_ADDR(hmodule));
+    module =
+        XModule::GetFromHModule(kernel_state(), (void*)hmodule.host_address());
```
Comment (DrChat) [inline src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc]: Again, no need to cast.

File: src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc [hunk 6]
```diff
@@ -21,7 +21,7 @@ namespace kernel {
 namespace xboxkrnl {
 
 X_STATUS xeExGetXConfigSetting(uint16_t category, uint16_t setting,
-                               void* buffer, uint16_t buffer_size,
+                               lpdword_t buffer, uint16_t buffer_size,
```
Comment (DrChat) [inline src/xenia/kernel/xboxkrnl/xboxkrnl_modules.cc]: Don't use a shim valuetype for a non-shim function

**Comment by maxton** [comment]
Thanks for the review, I've made the changes

**Comment by DrChat** [comment]
Nice, great work!

**Comment by Gh0stBlade** [comment]
This particular commit: c38accbb7650d920d3226360678ec56b72db6e5e makes Gears of War 3 worse, though it always displays a black screen. No music plays in the main menu past this commit. I even reverted those changed files on the latest build, recompiled, music is there.

I suspect there is a fault somewhere in c38accbb7650d920d3226360678ec56b72db6e5e which should be looked at.

**Pull request merged**